sentry/src/main/java/io/sentry/ApacheHttpClientTransportFactory.java

```diff
@@ -0,0 +1,55 @@
+package io.sentry;
+
+import io.sentry.transport.ApacheHttpClientTransport;
+import io.sentry.transport.ITransport;
+import io.sentry.transport.RateLimiter;
+import io.sentry.util.Objects;
+import org.apache.hc.client5.http.impl.DefaultConnectionKeepAliveStrategy;
+import org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClients;
+import org.apache.hc.client5.http.impl.nio.PoolingAsyncClientConnectionManager;
+import org.apache.hc.client5.http.impl.nio.PoolingAsyncClientConnectionManagerBuilder;
+import org.apache.hc.core5.http.impl.DefaultConnectionReuseStrategy;
+import org.apache.hc.core5.pool.PoolConcurrencyPolicy;
+import org.apache.hc.core5.util.TimeValue;
+import org.jetbrains.annotations.NotNull;
+
+/** Creates {@link ApacheHttpClientTransport}. */
+public final class ApacheHttpClientTransportFactory implements ITransportFactory {
```

Comment:
I'd say that we should extract this to a separate module and the other integrations (spring, spring-boot) would pull this dependency instead of the `sentry` dependency directly.
like `sentry-nio`, or something like that, by default `sentry` has the standard BlockingHttpTransport.

Reply:
Sounds good. I'll move it to another module

Reply:
maybe @bruno-garcia has opinions here too

Reply:
The module has been renamed.